libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -372,6 +376,10 @@ export class WorkspaceInstallManager extends BaseInstallManager {
   }
 
   protected async postInstallAsync(): Promise<void> {
+    if (this.twoStageInstall) {
+      this.commonTempInstallFlag.saveIfModified();
```

Comment:
The use of this flag file feels like it's getting muddled by this change. It used to mean "everything in the repo is installed and ready to use", now it means "some subset of the repo is installed and possibly the post-install scripts have been run".

It would make sense to use this file to do more specific state tracking (I like that you're including the scope in the file now). We should also care about running the post-install scripts though since not running them _could_ leave your repo in a less-usable state. Maybe we should check this file, and if `ignoreScripts` is `true` and `this.options.ignoreScripts` is false, we execute this stage, then write `ignoreScripts: false` in the flag file?

Reply:
Actually, there are only two cases: 1. `ignoreScripts: true` (deferredInstallationScripts or '--ignore-scripts' specified in CLI parameter); 2. In other cases, `ignoreScript` is missing in the flag file.

Reply:
It kind of complicated. Say we have run `rush install --ignore-scripts --to rush`, the flag file records:

```
"selectedProjects": ["<other_deps>", ..., "rush-lib", "rush"]
"ignoreScripts": true
```

Now, if i run `rush install --to rush-lib`, the stage 1 `pnpm install --ignore-scripts` should be skipped, since `--to rush-lib` is a subset of `--to rush`. And Rush just run stage 2 `pnpm rebuild --filter rush-lib...`. Now we can **not** mark `ignoreScripts: false` in the flag file. Since the `selectedProjects` contains "rush", which is out of range of `--filter rush-lib...`

One possible solution maybe is to record more information in one selected project in the flag file, use the above case as an example, the flag file will be

```
"selectedProjects": [
// ... other deps
{
  packageName: 'rush-lib',
  ignoreScripts: false,
},
{
  packageName: 'rush',
  ignoreScripts: true,  // <--- because we run "pnpm rebuild --pending --filter rush-lib..." which excludes "rush"
}
]
```

In this way, in `postinstallAsync` we need calculate `ignoreScripts` based on all selectedProjects to figure out whether we can skip "pnpm rebuild --pending"...

Do you think it's worth the efforts to implement this logic? For now, i choose an easy implementation which run "pnpm rebuild --pending [--filter <project...>]" everytime. `--pending` let pnpm helps decide whether there is any script we need to run.

Reply:
Good question. I think keeping more detailed info on the selected projects does make sense... it allows you to be very specific about the work you're performing and when you want to perform that work. Another option IMO would be to execute the install from scratch again (ie. running `pnpm install --ignore-scripts ...` then `pnpm rebuild ...`) whenever non-scoping arguments to the install have changed.

Basically, I feel like we either need to commit to tracking each project install on a per-project basis, or we need to treat all other options as something that could affect all projects and start from scratch.